core/spec/models/spree/stock/splitter/base_spec.rb

```diff
@@ -15,6 +15,22 @@ module Splitter
           splitter2.split(packages)
         end
 
+        it 'builds package using package factory' do
+          # Basic extension of Base splitter used to test build_package method
+          class ::BasicSplitter < Base
+            def split(packages)
+              build_package
+            end
+          end
+
+          # Custom package used to test setting package factory
+          class ::CustomPackage
+            def initialize(stock_location, order, splitters) end
+          end
+
+          packer.package_factory = CustomPackage
```

Comment:
For some reason I couldn't mock package_factory with either
`before { Spree::Config.package_factory = CustomPackage }`
or 
`allow(Spree::Config).to receive(:package_factory) { CustomPackage }`

I ended up making package_factory writable (attr_accessor instead of attr_reader above).

Reply:
could it be related to the fact that there is no `;` between the `initialize` and the `end` in line 28 above? AFAIK that is not valid Ruby.

Reply:
This seems to work:
```suggestion
          allow(Spree::Config).to receive(:package_factory) { CustomPackage }
```

Reply:
yeah, I was doing something wrong, it just works now...
@sauloperez the missing ; was not causing the problem but I added it anyway